atc/engine/check_delegate.go

```diff
@@ -125,7 +125,7 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 	interval := d.plan.Interval.Interval
```

Comment:
nit: Seems like I can't reach the line but on line 113, we can probably change 
```
} else if d.plan.Resource != "" {
```
to 
```
} else if d.plan.IsResourceCheck() {
```

Reply:
Yeap.